stages/apps/azure-example/main.tf

```diff
@@ -0,0 +1,75 @@
+provider "azurerm" {
+  version                    = "=2.20.0"
+  skip_provider_registration = true
+  features {}
+}
+
+provider "random" {
+  version = "=2.3.0"
+}
+
+data "azurerm_resource_group" "leadrg" {
+  name = var.resource_group
+}
+
+data "azurerm_subnet" "main" {
+  name = var.subnet_name
+  virtual_network_name = var.virtual_network_name
+  resource_group_name = var.resource_group
+}
+
+resource "random_password" "password" {
+  length           = 16
+  special          = true
+  override_special = "_%@"
+}
+
+resource "azurerm_network_interface" "main" {
+  name                = "${var.prefix}-nic1"
+  location            = data.azurerm_resource_group.leadrg.location
+  resource_group_name = var.resource_group
+
+  ip_configuration {
+    name                          = "leadconfiguration1"
+    subnet_id                     = data.azurerm_subnet.main.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_virtual_machine" "main" {
```

Comment:
are these meant to be example resources?  I wouldn't expect the `azurerm` provider to be used in the app stage.

Reply:
They are  examples. We were mainly trying to learn how to get information about existing resources to build another.